x-pack/plugins/siem/public/containers/source/index.tsx

```diff
@@ -42,27 +43,33 @@ interface WithSourceProps {
   sourceId: string;
 }
 
+const indexTypesVariables = [IndexType.ANY];
+
 export class WithSource extends React.PureComponent<WithSourceProps> {
   private memoizedIndexFields: (
     indexTypes: string[],
+    title: string,
     fields: IndexField[]
-  ) => StaticIndexPatternField[];
+  ) => StaticIndexPattern;
   private memoizedBrowserFields: (indexTypes: string[], fields: IndexField[]) => BrowserFields;
+  private memoizedIndexTypesLowerCase: (indexTypes: string[]) => string[];
 
   constructor(props: WithSourceProps) {
     super(props);
     this.memoizedIndexFields = memoize(this.getIndexFields);
     this.memoizedBrowserFields = memoize(this.getBrowserFields);
+    this.memoizedIndexTypesLowerCase = memoizeOne(this.getIndexTypesLowerCase);
   }
 
   public render() {
     const { children, sourceId, indexTypes = [IndexType.ANY] } = this.props;
+
```

Comment:
Above this you should want this here:

```ts
const { children, sourceId, indexTypes = indexTypesVariables } = this.props;
```

Because otherwise you are getting new reference arrays when you have defaults defined:
`indexTypes = [IndexType.ANY]`  

And then below `memoizeOne` usage won't be what you want. This case only happens when you have several defaults in a row.

Reply:
great catch!